awscli/examples/ec2/modify-image-attribute.rst

```diff
@@ -1,31 +1,54 @@
-**To make an AMI public**
+**Example 1: To make an AMI public**
 
-This example makes the specified AMI public. If the command succeeds, no output is returned.
+The following ``modify-instance-attribute`` example makes the specified AMI public. ::
 
-Command::
+    aws ec2 modify-image-attribute \
+        --image-id ami-5731123e \
+        --launch-permission "Add=[{Group=all}]"
 
-  aws ec2 modify-image-attribute --image-id ami-5731123e --launch-permission "Add=[{Group=all}]"
+This command produces no output.
 
-**To make an AMI private**
+**Example 2: To make an AMI private**
 
-This example makes the specified AMI private. If the command succeeds, no output is returned.
+The following ``modify-instance-attribute`` example makes the specified AMI private. ::
 
-Command::
+    aws ec2 modify-image-attribute \
+        --image-id ami-5731123e \
+        --launch-permission "Remove=[{Group=all}]"
 
-  aws ec2 modify-image-attribute --image-id ami-5731123e --launch-permission "Remove=[{Group=all}]"
+This command produces no output.
 
-**To grant launch permission to an AWS account**
+**Example 3: To grant launch permission to an AWS account**
 
-This example grants launch permissions to the specified AWS account. If the command succeeds, no output is returned.
+The following ``modify-instance-attribute`` example grants launch permissions to the specified AWS account. ::
 
-Command::
+    aws ec2 modify-image-attribute \
+        --image-id ami-5731123e \
+        --launch-permission "Add=[{UserId=123456789012}]"
 
-  aws ec2 modify-image-attribute --image-id ami-5731123e --launch-permission "Add=[{UserId=123456789012}]"
+This command produces no output.
 
-**To removes launch permission from an AWS account**
+**Example 4: To remove launch permission from an AWS account**
 
-This example removes launch permissions from the specified AWS account. If the command succeeds, no output is returned.
+The following ``modify-instance-attribute`` example removes launch permissions from the specified AWS account. ::
 
-Command::
+    aws ec2 modify-image-attribute \
+        --image-id ami-5731123e \
+        --launch-permission "Remove=[{UserId=123456789012}]"
 
-  aws ec2 modify-image-attribute --image-id ami-5731123e --launch-permission "Remove=[{UserId=123456789012}]"
+**Example 5: To modify the user data attached to an instance**
+
+The following ``modify-instance-attribute`` example shows how to add the contents of a file to the UserData for an instance. Note that the Value string must be the actual string value; you cannot use the file:// syntax to pass the file. If you use this technique, you do not have to base64 encode the file's contents first. ::
+
+    aws ec2 modify-image-attribute \
+        --instance-id=i-09b5a14dbca622e76 \
+        --user-data Value="$(UserData.txt)"
```

Comment:
For the `Value` part of this snippet, I'm not sure if that would work. Wouldn't we have to use `cat` to get the contents of the file?

Reply:
That example was actually in the wrong command.  I moved it to modify-INSTANCE-attribute, and corrected the syntax.  I ran it to confirm that it runs correctly.